### PR TITLE
backport: dashmap change to fix security issue

### PR DESCRIPTION
**Pull request by TommyCpp**
Backport dashmap change in #818 to v0.17.0

### CODE REVIEW

File: opentelemetry/Cargo.toml [hunk 1]
```diff
@@ -22,7 +22,7 @@ rustdoc-args = ["--cfg", "docsrs"]
 [dependencies]
 async-std = { version = "1.6", features = ["unstable"], optional = true }
 async-trait = { version = "0.1", optional = true }
-dashmap = { version = "4.0.1", optional = true }
+dashmap = { version = "=5.1.0", optional = true }
```
Comment (djc) [inline opentelemetry/Cargo.toml:25]: Maybe add a comment here about why we're pinning?

File: examples/external-otlp-tonic-tokio/src/main.rs [hunk 2]
```diff
@@ -47,7 +47,7 @@ fn init_tracer() -> Result<sdktrace::Tracer, TraceError> {
         .map(|(name, value)| {
             let header_name = name
                 .strip_prefix(HEADER_PREFIX)
-                .map(|h| h.replace("_", "-"))
+                .map(|h| h.replace('_', "-"))
```
Comment (djc) [inline examples/external-otlp-tonic-tokio/src/main.rs:50]: It's a little ugly to have this in a commit whose message is "change log".

File: opentelemetry-aws/CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## v0.5.1
```
Comment (jtescher) [inline opentelemetry-aws/CHANGELOG.md:3]: Do we need to release new versions of these dependent crates? they should still be compatible with the 0.17.1 without a change right?

Comment (TommyCpp) [reply opentelemetry-aws/CHANGELOG.md:3]: Yep no user facing API has changed. So probably we don't need release a new version for each crates. I will remove those

File: examples/external-otlp-tonic-tokio/src/main.rs (hunk 2 above)
Comment (TommyCpp) [reply examples/external-otlp-tonic-tokio/src/main.rs:50]: Sorry I didn't clean up the commit logs. Will force push a cleaned up version

**Comment by TommyCpp** [comment]
MSRV issue is caused by https://github.com/plotters-rs/plotters/issues/398.

**Comment by TommyCpp** [comment]
I think with the new release we can close this out

**Pull request closed without merging**